pandas/_libs/tslib.pyx

```diff
@@ -609,10 +612,12 @@ cpdef array_to_datetime(
                 val = values[i]
                 if is_integer_object(val) or is_float_object(val):
                     result[i] = NPY_NAT
+        elif allow_mixed:
+            pass
```

Comment:
can we just handle here, this makes the caller very messy.

Reply:
depends in large part on why we dont allow_mixed ATM.  We want to change the behavior in one place this is called but i need to check the other callers to make sure allow_mixed wont break their worlds.

I agree it is messy, am trying to unify a bunch of these things.  xref #40110, #40103